### DISCUSSION TITLE
callback directly or process.nextTick callback. why

### DISCUSSION THREAD

**Discussion by wmvm0** [sidorares/node-mysql2#1547] [Q&A]
Why not call callback directly, but use process.nextTick？can you tell me what you think

```js
class Query extends Command {
  done() { 
      process.nextTick(() => {
          this.onResult(null, rows, fields);
     });
}
```

**Comment by sidorares** [comment]
I believe back years ago that was a more performant way compared to `try { this.onResult(...) } catch {}`
Not so sure if this is still the case
